plugins/modules/azure_rm_monitoractiongroups.py

```diff
@@ -0,0 +1,754 @@
+        action_group_template = {
+            "location": self.location,
+            "tags": self.tags,
+            "group_short_name": self.group_short_name,
+            "enabled": self.enabled,
+            "email_receivers": self.email_receivers,
+            "sms_receivers": self.sms_receivers,
+            "webhook_receivers": self.webhook_receivers,
+            "itsm_receivers": self.itsm_receivers,
+            "azure_app_push_receivers": self.azure_app_push_receivers,
+            "automation_runbook_receivers": self.automation_runbook_receivers,
+            "voice_receivers": self.voice_receivers,
+            "logic_app_receivers": self.logic_app_receivers,
+            "azure_function_receivers": self.azure_function_receivers,
+            "arm_role_receivers": self.arm_role_receivers,
+            "event_hub_receivers": self.event_hub_receivers,
+        }
+        # Filter out all None values
+        action_group_input = {key: value for key, value in action_group_template.items() if value is not None}
+
+        # Create/Update if state==present
+        if self.state == 'present':
+            if before_dict is None:
+                # action group does not exist, create
+                # On creation default to location of resource group unless otherwise noted in input variables
+                if not self.location:
+                    resource_group = self.get_resource_group(self.resource_group)
+                    action_group_input['location'] = resource_group.location
+                # On creation input == what we send to api
+                action_group_update = action_group_input
+                self.results['changed'] = True
+                if self.check_mode:
+                    # Check mode, skipping actual creation
+                    pass
+                else:
+                    create_response = self.create_or_update(action_group_update)
+            else:
+                # action group already exists, updating it
+                # Dict for update is the union of existing object overwritten by input data
+                action_group_update = before_dict | action_group_input
+
+                # Enhanced with tags (special behaviour because of append_tags possibility)
+                update_tags, update_tags_content = self.update_tags(before_dict.get('tags'))
+                if update_tags:
+                    action_group_update['tags'] = update_tags_content
+                    self.results['changed'] = True
+                    if self.check_mode:
+                        # Check mode, skipping actual creation
+                        pass
+                    else:
+                        create_response = self.create_or_update(action_group_update)
+                # Check if we need to update the action group if not already updated because of tags
+                elif not self.default_compare({}, action_group_update, before_dict, '', result_compare):
```

Comment:
The following also needs to be changed in this way. If append_tags = True, self.tags is a subset of before.get('tags'), and not default_compare is True, it will overwrite tags. Thank you!
```suggestion
                else:
                    action_group_update['tags'] = update_tags_content
                    if  not self.default_compare({}, action_group_update, before_dict, '', result_compare):
                        ********
```

Reply:
Both #1971 and #1952 should make the same changes. Thank you!